pkg/controllers/mcs/serviceexport_controller.go

```diff
@@ -138,6 +138,28 @@ func (c *SeController) Handle(obj interface{}) (requeueAfter *time.Duration, err
 		return nil, err
 	}
 
+	// remove endpoint slices exist in parent but not in child cluster
+	localEndpointSliceMap := make(map[string]bool, 0)
+	for _, item := range endpointSliceList {
+		localEndpointSliceMap[fmt.Sprintf("%s-%s", se.Namespace, item.Name)] = true
+	}
+	if parentEndpointSliceList, err := c.parentk8sClient.DiscoveryV1().EndpointSlices(c.dedicatedNamespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{discoveryv1.LabelServiceName: utils.DerivedName(namespace, seName)}).String()}); err == nil {
+		shouldRetry := false
```

Comment:
It seems `shouldRetry` is redundant. We should return non-nil `err`.

Reply:
ok